### PR TITLE
RavenDB-22327 Enhanced logging clarity for `Responsible Node` information when stopping ETL Task. Added explanation for swiching a task to a different node.

### PR DESCRIPTION
**Pull request by arekpalinski**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-22327/Enhance-logging-clarity-for-Responsible-Node-information-in-ETL-Task-stoppage

### Additional description

Including info about actual reason of moving a task to a different node.

### Type of change

- [ ] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [ ] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [x] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [ ] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

File: src/Raven.Client/ServerWide/DatabaseTopology.cs [hunk 1]
```diff
@@ -442,29 +442,51 @@ public void RemoveFromTopology(string delDbFromNode)
...
             if (lastResponsibleNode != null)
+            {
+                explanations?.Add($"Task belongs to last responsible node ({lastResponsibleNode})");
                 return lastResponsibleNode;
+            }
+
+            var node = WhoseTaskIsIt(task, explanations);
+
+            if (node != null)
+                explanations?.Add($"Task belongs to node {node}");
```
Comment (ppekrol) [inline src/Raven.Client/ServerWide/DatabaseTopology.cs]: maybe we want here to lag that node is null as well?

**Review by ppekrol — COMMENTED** [review]
I think this will require a separate PR to 6.0

File: src/Raven.Client/ServerWide/DatabaseTopology.cs (hunk 1 above)
Comment (arekpalinski) [reply src/Raven.Client/ServerWide/DatabaseTopology.cs]: done

**Comment by arekpalinski** [comment]
> I think this will require a separate PR to 6.0

6.0 PR: https://github.com/ravendb/ravendb/pull/19153

**Pull request merged**